tests/phpunit/CRM/Price/BAO/PriceFieldTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 5                                                  |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2019                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * Test class for CRM_Price_BAO_PriceField.
+ * @group headless
+ */
+class CRM_Price_BAO_PriceFieldTest extends CiviUnitTestCase {
+
+  /**
+   * Sets up the fixtures.
+   */
+  protected function setUp() {
+    parent::setUp();
+  }
+
+  /**
+   * Tears down the fixture.
+   */
+  protected function tearDown() {
+  }
+
+
+  public function testSubmitPriceFieldWithOptions() {
+    $this->priceSet = civicrm_api3('PriceSet', 'create', [
+      'is_active' => 1,
+      'extends' => 2,
+      'is_quick_config' => 1,
+      'financial_type_id' => 1,
+      'name' => 'test_price_set',
+      'title' => 'Test Price Set',
+    ]);
+    $this->priceFieldParams = [
+      'name' => 'contribution_amount',
+      'is_active' => 1,
+      'weight' => 2,
+      'is_required' => 1,
+      'label' => 'Contribution Amount',
+      'html_type' => 'Radio',
+      'financial_type_id' => 1,
+      'option_label' => [
```

Comment:
This is the strange format that is used when saving the fees / options from the UI